### PR TITLE
`http.write` block #337

### LINKED ISSUE #337: `http.write` block

### PR DESCRIPTION
**Pull request by spicy-sauce**

### CODE REVIEW

File: core/src/datayoga_core/blocks/http/write/block.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                elif self.method == "POST":
+                    response = requests.post(url, data=payload, params=query_params,
+                                             headers=headers, timeout=self.timeout)
+                elif self.method == "PUT":
+                    response = requests.put(url, data=payload, params=query_params,
+                                            headers=headers, timeout=self.timeout)
+                elif self.method == "DELETE":
+                    response = requests.delete(url, params=query_params, headers=headers, timeout=self.timeout)
+
+                if response:
```
Comment (vd2org) [inline core/src/datayoga_core/blocks/http/write/block.py]: Could the `response` be false?

**Review by vd2org — CHANGES_REQUESTED** [review]

File: core/src/datayoga_core/blocks/http/write/block.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            try:
+                url = f"{self.base_uri}/{request_configs['endpoint'][i].lstrip('/')}"
+                headers = {key: value[i] for key, value in request_configs["headers"].items()}
+                query_params = {key: value[i] for key, value in request_configs["query_params"].items()}
+                payload = {key: value[i] for key, value in request_configs["payload"].items()}
+
+                logger.debug(
+                    f"Sending HTTP {self.method} request to: {url}\nheaders:{headers}\n\tquery_params: {query_params}\n\tpayload: {payload}")
+
+                if self.method == "GET":
```
Comment (vd2org) [inline core/src/datayoga_core/blocks/http/write/block.py]: And… why not `aiohttp` or `httpx` which are async ?

**Review by vd2org — CHANGES_REQUESTED** [review]

File: core/src/datayoga_core/blocks/http/write/block.py (hunk 1 above)
Comment (spicy-sauce) [reply core/src/datayoga_core/blocks/http/write/block.py]: Replaced with aiohttp, so irrelevant.

File: core/src/datayoga_core/blocks/http/write/block.py (hunk 2 above)
Comment (spicy-sauce) [reply core/src/datayoga_core/blocks/http/write/block.py]: Done.

**Review by vd2org — APPROVED** [review]

**Pull request merged**